src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -521,10 +1051,13 @@ public void Abort_BeginGetRequestStreamThenAbort_EndGetRequestStreamThrowsWebExc
 
             request.BeginGetResponse(new AsyncCallback(RequestStreamCallback), state);
 
-            request.Abort();
-            Assert.Equal(1, state.RequestStreamCallbackCallCount);
-            WebException wex = state.SavedRequestStreamException as WebException;
-            Assert.Equal(WebExceptionStatus.RequestCanceled, wex.Status);
+            if (!request.HaveResponse)
```

Comment:
What is the "by design" according to MSDN?  What could deadlock here?

There's certainly a race condition with regards to whether the request will have completed by the time Abort is called; nothing to do about that.  But that just means the checks for whether the operation completes successfully or not should accommodate the potential of the request succeeding or being canceled.  Adding the check you've added only decreases the chances of hitting the race condition, but doesn't remove it entirely.

And to do this actually correctly, rather than going against a remote server we don't control, we should really be going against a local loopback server such that we can guarantee the abort will be requested before the operation has completed.